docs/pages/_app.js

```diff
@@ -75,6 +76,7 @@ ponyfillGlobal.muiDocConfig = {
       '@mui/x-data-grid-pro': getMuiPackageVersion('x-data-grid-pro', muiCommitRef),
       '@mui/x-data-grid-generator': getMuiPackageVersion('x-data-grid-generator', muiCommitRef),
       '@mui/x-data-grid': getMuiPackageVersion('x-data-grid', muiCommitRef),
+      '@mui/x-date-pickers': getMuiPackageVersion('x-date-pickers', muiCommitRef),
```

Comment:
`versions['@mui/x-date-pickers']` was not defined, so it was reseting the value

Reply:
You also need to set the version for `@mui/x-date-pickers-pro`. Look https://codesandbox.io/s/kd23vd?file=/package.json It's always `latest`.